autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -175,7 +175,15 @@ def update_execution(status: ExecutionStatus) -> NodeExecutionResult:
     input_data, error = validate_exec(node, data.data, resolve_input=False)
     if input_data is None:
         log_metadata.error(f"Skip execution, input validation error: {error}")
-        db_client.upsert_execution_output(node_exec_id, "error", error)
+        _register_node_execution_output(
```

Comment:
You can create a local function like `update_execution` above to avoid passing a large number of variables into your provide method thrice. Only output_name and output_data changed; the status for update_execution can be inferred. And the `update_execution` can be just inlined in this new local function.